pkg/cmd/create/create.go

```diff
@@ -751,6 +783,28 @@ func (o *Options) watchNamespaceStart() error {
 	return nil
 }
 
+func (o *Options) IfPodIsFailedShareLogs(pod *corev1.Pod, previewNamespace string) error {
+	if pod.Status.Phase == corev1.PodFailed {
+		log.Logger().Infof("found pod %s in namespace %s in state %s", pod.Name, previewNamespace, pod.Status.Phase)
+
+		logs := o.KubeClient.CoreV1().Pods(previewNamespace).GetLogs(pod.Name, &corev1.PodLogOptions{})
+		stream, err := logs.Stream(context.Background())
+		if err != nil {
+			return err
+		}
+		defer stream.Close()
+
+		scanner := bufio.NewScanner(stream)
+
+		var m string
+		for scanner.Scan() {
+			m = scanner.Text()
+		}
+		return fmt.Errorf("failed pod %s in namespace %s: %s", pod.Name, previewNamespace, m)
```

Comment:
Hey, I just tried to work out where that came from in my brain, that's exactly how the kubectl get event system works, same input. So I'm confident this will work as we expect.

https://github.com/jenkins-x-plugins/jx-preview/blob/feat/pod_logs_for_time_out/pkg/cmd/create/create.go#L771

https://github.com/jenkins-x-plugins/jx-preview/blob/feat/pod_logs_for_time_out/pkg/cmd/create/create.go#L779

Reply:
But that is very different since m is used in a log statement and not returned.